ptp/ptp4u/server/server_test.go

```diff
@@ -121,3 +122,28 @@ func TestSendGrant(t *testing.T) {
 
 	s.sendGrant(sc, &ptp.Signaling{}, 0, 0, 0, sa)
 }
+
+func TestDrain(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	s := Server{
+		ctx:    ctx,
+		cancel: cancel,
+	}
+
+	require.NoError(t, s.ctx.Err())
+	s.Drain()
+	require.ErrorIs(t, s.ctx.Err(), context.Canceled)
```

Comment:
expected argument is always first :)

Reply:
I always confuse the order, let me swap them

Reply:
Updated